server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -1317,11 +1317,11 @@ public void failEngine(String reason, @Nullable Exception failure) {
                         }
                     }
 
-                    // If cleanup of unreferenced flag is enabled and force merge or regular merge failed due to IOException,
-                    // clean all unreferenced files on best effort basis created during failed merge and reset the
-                    // shard state back to last Lucene Commit.
-                    if (shouldCleanupUnreferencedFiles() && isMergeFailureDueToIOException(failure, reason)) {
-                        logger.info("Cleaning up unreferenced files as merge failed due to: {}", reason);
+                    // If cleanup of unreferenced file flag is enabled and any operation failed due to IOException,
+                    // clean up all unreferenced files on best effort basis created during failed merge and reset the
+                    // shard state back to last Lucene Commit
+                    if (shouldCleanupUnreferencedFiles() && isOperationFailureDueToIOException(failure)) {
+                        logger.info("Cleaning up unreferenced files created during failed merge due to: {}", reason);
```

Comment:
Should be warning?

Reply:
We would be doing extra clean ups on may be other shards as well due to this . Please evaluate if we can avoid the same and do clean up only on the responsible shard instead